packages/plugin-core/src/survey.ts

```diff
@@ -0,0 +1,173 @@
+import { SurveyEvents } from "@dendronhq/common-all";
+import { AnalyticsUtils } from "./utils/analytics";
+import * as vscode from "vscode";
+import _ from "lodash";
+
+export class DendronSurvey {
+  choices: readonly vscode.QuickPickItem[];
+  opts: {
+    placeHolder: string;
+    canPickMany: boolean;
+    title: string;
+    ignoreFocusOut: boolean;
+  };
+
+  constructor(opts: {
+    choices: vscode.QuickPickItem[];
+    canPickMany: boolean;
+    title: string;
+    placeHolder?: string;
+  }) {
+    const { choices, canPickMany, title } = opts;
+    let placeHolder = opts.placeHolder;
+    this.choices = choices;
+
+    if (!placeHolder) {
+      placeHolder = canPickMany ? "Check all that applies." : "Check one";
+    }
+
+    this.opts = { title, placeHolder, canPickMany, ignoreFocusOut: true };
+  }
+
+  getChoices(): readonly vscode.QuickPickItem[] {
+    return this.choices;
+  }
+
+  onAnswer(_opts: any): void {
+    return undefined;
+  }
+  onReject(_opts?: any): void {
+    return undefined;
+  }
+
+  async show() {
+    const results = await vscode.window.showQuickPick(this.choices, this.opts);
+    if (results) {
+      this.onAnswer(results);
+    } else {
+      this.onReject();
+    }
+
+    return results;
+  }
+}
+
+export class BackgroundSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.BackgroundAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.BackgroundRejected);
+  }
+
+  static create() {
+    const title = "What is your background?";
+    const choices = [
+      { label: "Software Developer" },
+      { label: "Technical Writer" },
+      { label: "Dev Ops" },
+      { label: "Manager" },
+      { label: "Student" },
+      { label: "Other" },
+    ];
+    return new BackgroundSurvey({ title, choices, canPickMany: true });
+  }
+}
+
+export class UseCaseSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.UseCaseAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.UseCaseRejected);
+  }
+
+  static create() {
+    const title = "Tell us what you want to use Dendron for";
```

Comment:
nit: maybe for extra clarity that they can select multiple, and maybe drop 'tell us': 

"How would you like to use Dendron? Select all that apply"

Also - it'd be good to maybe add into the title or somewhere to show the progress - that they're on question 1/3, 2/3, 3/3 - that way, they know it's very short and that they'll be done in < 1 minute.

Reply:
Good point. will change

Reply:
It's pretty clear that you can select multiples though.
![image](https://user-images.githubusercontent.com/1219789/134870486-d6cca67a-54d9-467b-bba6-5cec30dcf2b3.png)

Reply:
Cool - in that case, no need.

Reply:
I did clean up the message and add steps / total as you suggested.